snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -249,6 +250,32 @@ def is_clean(self, step):
         except errors.NoLatestStepError:
             return True
 
+    def is_outdated(self, step):
+        """Return true if the given step needs to be updated (no cleaning)."""
+
+        return self.get_outdated_report(step) is not None
+
+    def get_outdated_report(self, step: steps.Step):
+        """Return an OutdatedReport class describing why step is outdated.
```

Comment:
Can you add a follow up paragraph on what it means to be outdated? Simil for what it means to be dirty in its counterpart.

Reply:
Done, for both outdated and dirty.